src/lib/viewers/image/ImageBaseViewer.js

```diff
@@ -205,15 +205,15 @@ class ImageBaseViewer extends BaseViewer {
                             const aspectRatio = h ? w / h : w;
                             imageEl.setAttribute('originalWidth', Math.round(aspectRatio * 150));
                             imageEl.setAttribute('originalHeight', 150);
-                            resolve();
                         } catch (e) {
                             // Assume 300x150 that chrome does by default
                             imageEl.setAttribute('originalWidth', 300);
                             imageEl.setAttribute('originalHeight', 150);
-                            resolve(e);
+                        } finally {
+                            resolve();
                         }
                     })
-                    .catch(reject);
+                    .catch(resolve);
```

Comment:
What does resolving in this catch do? Also is there a reason this is all async? Shouldn't everything dimension-wise be available after the image is done loading?

Reply:
The get call is async so we need to know when it resolves in order to proceed. Otherwise zoom will get called in finishloading() and the scale will be set to infinity. Sometimes the get call is not called (original dimensions are set) and the promise is resolved right away. On any errors, we should resolve so that the image still loads (although with an incorrect data-scale attribute).